### ISSUE TITLE
cppcoreguidelines-pro-type-member-init triggers for system types

### ISSUE THREAD

**Issue by llvmbot** [llvm/llvm-project#30768]
|  |  |
| --- | --- |
| Bugzilla Link | [31420](https://llvm.org/bz31420) |
| Version | unspecified |
| OS | Linux |
| Reporter | LLVM Bugzilla Contributor |

## Extended Description 
// Does not make sense to warn here because API requires sigemptyset() to be called.
struct sigaction action;
sigemptyset(&action.sa_mask);

...

// Does not make sense to warn here because ioctl() only returns success if struct is filled out.
struct winsize ws;
if (ioctl(STDOUT_FILENO, TIOCGWINSZ, &ws) == -1)
    ....

User cannot add constructors to these types that zero initialses all members. It is possible to sprinkle "= {}" everywhere but would it not simply be better to ignore structs in system headers that the user does not control?

**Comment by llvmbot** [comment]
I also see false positives when using std::mutex as an instance variable in classes.

```
foo.h:29:8: warning: constructor does not initialize these fields: mutex [cppcoreguidelines-pro-type-member-init]
        class Foo
```

And Foo looks like this:

```c++
class Foo
{
public:
    virtual ~Foo();
    ...

private:
    std::mutex mutex;
    ...
};
```

No short code snippet that illustrates the problem but Foo does not contain much more than that.

**Comment by vbvictor** [comment]
https://godbolt.org/z/o4d8hbjT4
